src/qibo/quantum_info/superoperator_transformations.py

```diff
@@ -218,6 +220,130 @@ def to_liouville(channel, order: str = "row", backend=None):
     return channel
 
 
+def hamming_weight_array(arr, backend):
+    """
+    Vectorized Hamming weight computation for arrays of integers.
+    Args:
+        arr: Integer array.
+        backend: Qibo backend.
+    Returns:
+        Array with the same shape, containing the Hamming weight of each element.
+    """
+    arr = backend.np.array(arr, dtype=backend.np.uint64)
+    weights = backend.np.zeros_like(arr)
+    while backend.np.any(arr):
+        weights += arr & 1
+        arr >>= 1
+    return weights
```

Comment:
What's exactly the input of this function? It seems an extra complication when we already have an implementation for the Hamming weight function.